tests/Unit/Core/Cart/AbstractCartTest.php

```diff
@@ -299,22 +299,30 @@ protected function insertCartRule($cartRuleFixtureId, $cartRuleData)
 
     protected function addCartRulesToCart(array $cartRuleFixtureIds)
     {
+        $allAdded = true;
         foreach ($cartRuleFixtureIds as $cartRuleFixtureId) {
             $cartRule = $this->getCartRuleFromFixtureId($cartRuleFixtureId);
-            if ($cartRule !== null) {
+            if ($cartRule === null) {
+                $allAdded = false;
```

Comment:
Maybe break when $allAdded is set to false or does it need to add others?

Reply:
I added a comment : here we just silent not-added cart rules. But some tests have to know the global result